packages/babel-helper-remap-async-to-generator/src/index.js

```diff
@@ -14,11 +14,24 @@ let buildWrapper = template(`
   })
 `);
 
+let arrowBuildWrapper =  template(`
+  (() => {
+    var ref = FUNCTION;
+    return (PARAMS) => ref.apply(this, arguments);
```

Comment:
Yes, it should to bind on outer scopes `this`. Because FUNCTION is arrow function originally, it should not bind on wrapper scope as it happens in the current implementation. So I used arrow function for inheritance of outer scopes `this` by wrapper. I don't know a better way to do it (I'm newbie at babel and there is a few docs). I'm open to suggestions.
And what else I'm not entirely sure about some nested arrow/regular functions.